### PR TITLE
cmd_reboot

### PR DESCRIPTION
**Pull request by AgnieszkaBylica**

### CODE REVIEW

File: moler/cmd/unix/reboot.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from moler.cmd.unix.genericunix import GenericUnixCommand
+
+
+class Reboot(GenericUnixCommand):
+
+    def __init__(self, connection, prompt=None, newline_chars=None, runner=None):
+        """
+        :param connection: connection
+        :param prompt: Prompt of the starting shell
+        :param newline_chars: end of line characters
```
Comment (marcin-usielski) [inline moler/cmd/unix/reboot.py]: Missing 'runner' description.

File: moler/cmd/unix/reboot.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        """
+
+        super(Reboot, self).__init__(connection=connection, prompt=prompt, newline_chars=newline_chars, runner=runner)
+
+    def build_command_string(self):
+        return "reboot"
+
+    def on_new_line(self, line, is_full_line):
+        self.set_result({})
```
Comment (marcin-usielski) [inline moler/cmd/unix/reboot.py]: Flag  ret_required = False is not enough in this case?

Comment (AgnieszkaBylica) [reply moler/cmd/unix/reboot.py]: No, because with ret_required=False I get CommandTimeout.

Comment (AgnieszkaBylica) [reply moler/cmd/unix/reboot.py]: Change of plans: now I catch closed ssh connection and I set result.

Comment (Ernold11) [reply moler/cmd/unix/reboot.py]: Ok but what if connection will be telnet? If the message is the same?

Comment (tkrol88) [reply moler/cmd/unix/reboot.py]: I added patterns for telnet and console connection. Could you review again?

File: moler/cmd/unix/reboot.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        """
+        if is_full_line:
+            self._catch_connection_closed(line)
+        else:
+            self._catch_login_prompt(line)
+
+    _re_connection_closed = re.compile(r"(?P<CLOSED>(Connection\s+to\s+\S+\s+closed.*)|"
+                                       r"(Connection closed by foreign host.*))", re.I)
+
+    _re_login_prompt = re.compile(r"(?P<CLOSED>.*?login:\s+)", re.I)
```
Comment (Ernold11) [inline moler/cmd/unix/reboot.py]: Please rename regex group name. I think it should be e.g. LOGIN_PROMPT.

Comment (tkrol88) [reply moler/cmd/unix/reboot.py]: done

**Pull request merged**